### PR TITLE
Park Vehicle Refactoring

### PR DESCRIPTION
**Pull request by johnkallstrom**
- Add CreateParkedVehicleViewModel
- Dropdown for Garages on Park Vehicle View
- Add AlertViewModel
- Add IndexParkedVehicleViewModel
- Alert feedback when parking a new vehicle

### CODE REVIEW

File: Garage_2_0/Controllers/VehicleController.cs [hunk 1]
```diff
@@ -2,18 +2,24 @@
...
+        private readonly IRepository<ParkedVehicle> _vehicleRepository = vehicleRepository;
 
         public async Task<IActionResult> Index()
         {
-            var vehicles = await _repository.All();
+            var vehicles = await _vehicleRepository.All();
 
-            var model = vehicles.Select(x => new ParkedVehicleSlimViewModel
+            var model = vehicles
+                .OrderByDescending(x => x.RegisteredAt)
```
Comment (suny-am) [inline Garage_2_0/Controllers/VehicleController.cs:21]: Prefer descriptive variables: 'x' => 'v'

File: Garage_2_0/Controllers/VehicleController.cs [hunk 2]
```diff
@@ -28,7 +34,7 @@ public async Task<IActionResult> Index()
 
         public async Task<IActionResult> Details(int id)
         {
-            var result = await _repository.Find(v => v.Id == id);
+            var result = await _vehicleRepository.Find(v => v.Id == id);
```
Comment (suny-am) [inline Garage_2_0/Controllers/VehicleController.cs:37]: Prefer descriptive variables: 'x' => 'v'

File: Garage_2_0/Controllers/VehicleController.cs [hunk 3]
```diff
@@ -44,14 +50,25 @@ public async Task<IActionResult> Details(int id)
...
 
-        public IActionResult Create()
+        public async Task<IActionResult> Create()
         {
-            return View();
+            var garages = await _garageRepository.All();
+
+            var viewModel = new CreateParkedVehicleViewModel
+            {
+                Garages = garages.Select(x => new SelectListItem
```
Comment (suny-am) [inline Garage_2_0/Controllers/VehicleController.cs:59]: Prefer descriptive variables: 'x' => 'v'

File: Garage_2_0/Controllers/VehicleController.cs [hunk 4]
```diff
@@ -66,9 +83,24 @@ public async Task<IActionResult> Create(ParkedVehicle viewModel)
                     Color = viewModel.Color
                 };
 
-                await _repository.Create(vehicle);
+                vehicle.RegisteredAt = DateTime.Now;
+
+                var result = await _garageRepository.Find(x => x.ID == viewModel.GarageId);
```
Comment (suny-am) [inline Garage_2_0/Controllers/VehicleController.cs:88]: `(await _garageRepository.Find(g => g.ID == viewModel.GarageId)).Single();`

File: Garage_2_0/Controllers/VehicleController.cs [hunk 5]
```diff
@@ -2,18 +2,24 @@
...
+        private readonly IRepository<Garage> _garageRepository = garageRepository;
+        private readonly IRepository<ParkedVehicle> _vehicleRepository = vehicleRepository;
 
         public async Task<IActionResult> Index()
         {
-            var vehicles = await _repository.All();
+            var vehicles = await _vehicleRepository.All();
 
-            var model = vehicles.Select(x => new ParkedVehicleSlimViewModel
+            var model = vehicles
```
Comment (suny-am) [inline Garage_2_0/Controllers/VehicleController.cs:20]: Pluralize variable: 'model' => 'viewModels'

File: Garage_2_0/Controllers/VehicleController.cs [hunk 6]
```diff
@@ -44,14 +50,25 @@ public async Task<IActionResult> Details(int id)
             return View(model);
         }
 
-        public IActionResult Create()
+        public async Task<IActionResult> Create()
         {
-            return View();
+            var garages = await _garageRepository.All();
+
+            var viewModel = new CreateParkedVehicleViewModel
```
Comment (suny-am) [inline Garage_2_0/Controllers/VehicleController.cs:57]: Pluralize variable: 'viewModel' => 'viewModels'

**Review by suny-am — CHANGES_REQUESTED** [review]
Mostly just syntax stuff!

File: Garage_2_0/Controllers/VehicleController.cs (hunk 6 above)
Comment (johnkallstrom) [reply Garage_2_0/Controllers/VehicleController.cs:57]: This is just one single viewmodel for creating/parking a new vehicle, don't see why this needs to be in pluralform

**Comment by johnkallstrom** [comment]
All comments and merge conflicts should be solvd  now

**Review by suny-am — APPROVED** [review]
🎖️

**Pull request merged**